### DISCUSSION TITLE
Welcome to hugo-embed-pdf-shortcode Discussions!

### DISCUSSION THREAD

**Discussion by anvithks** [anvithks/hugo-embed-pdf-shortcode#24] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sysadmin-info** [comment]
Hi,

I was wondering is there any option to add i18n translation to at least translate buttons next and previous. Of course I can try to do it on my own. I know I can hardcode this oj file, but you know that this approach is not something that I want to achieve. Besides adding a language file is much easier for average user.

**Reply by anvithks** [reply]
@sysadmin-info I have not really thought about this so far.
Hard-coding the translations will not be scalable for all languages.
I will have to check how we can do this in a shortcode.
Any help in this will be much appreciated.

**Reply by sysadmin-info** [reply]
The solution is quite simple. All you have to do is to create a i18n directory inside your plugin, where README.md is located. Then put inside the i18n directory file en.toml, pl.toml, fr.toml etc. for each language. 

Example content of en.toml
```
[embed-pdf-next]
other = "Next"

[embed-pdf-prev]
other = "Previous"
```

Example content of pl.toml
```
[embed-pdf-next]
other = "Następna"

[embed-pdf-prev]
other = "Poprzednia"
```

Then in your file embed-pdf.html inside the shortcodes directory you have to change Next and Previous to this:

```
        <button id="pdf-prev-{{ substr (.Get "url" | md5) 0 8 }}">{{ i18n "embed-pdf-prev" }}</button>
        <button id="pdf-next-{{ substr (.Get "url" | md5) 0 8 }}">{{ i18n "embed-pdf-next" }}</button>
```
